test/redpanda/mzworkflows.py

```diff
@@ -0,0 +1,36 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from materialize.mzcompose import Materialized, Redpanda, Testdrive, Workflow
+
+prerequisites = [Redpanda(), Materialized()]
+
+services = [
+    *prerequisites,
+    Testdrive(shell_eval=True, volumes_extra=["../testdrive:/workdir/testdrive"]),
+]
+
+
+def workflow_redpanda_testdrive(w: Workflow):
+    w.start_and_wait_for_tcp(services=prerequisites)
+    w.wait_for_mz(service="materialized")
+
+    # Features currently not supported by Redpanda:
+    # - `kafka-time-offset.td` (https://github.com/vectorizedio/redpanda/issues/2397)
+    # - `schema-registry-publish` with `format=protobuf` (Redpanda does not support schema publication for protobuf/json)
+
+    # Due to interactions between docker-compose, entrypoint, command, and bash, it is not possible to have
+    # a more complex filtering expression in 'command' . So we basically run the entire testdrive suite here
+    # except tests that contain features known to be not supported by Redpanda. So the run includes testdrive
+    # tests that do not touch Kafka at all.
```

Comment:
Well, it is Python, so you could just execute whatever logic you want ahead of time to construct the list of files! You could `spawn.capture(["bash", "-c", "grep", "..."]))` for example.

Reply:
Yes, I have been trying to figure out how to handle this in a unified way across tests and I have been swinging between letting python process TD_TEST and expand it all the way to having testdrive handle TD_TEST internally ... so far each test  has required a different thing so I am still struggling to find a common pattern.